rllib/env/single_agent_env_runner.py

```diff
@@ -274,7 +274,7 @@ def _sample_timesteps(
                 if explore:
                     env_steps_lifetime = self.metrics.peek(
                         NUM_ENV_STEPS_SAMPLED_LIFETIME
```

Comment:
`NUM_ENV_STEPS_SAMPLED_LIFETIME` is then update during synching the metrics in between sampling?

Reply:
Correct, it's always updated anyways (this was a bug) after each single env step. Inside `self._increase_sampled_metrics()`.

Reply:
And it's also hard-overwritten by the Algo itself after each iteration in order to sum up all the steps from all the other EnvRunners (making sure each EnvRunner always has the total steps sampled, not just its own lifetime count).